rotkehlchen/chain/evm/decoding/merkl/decoder.py

```diff
@@ -0,0 +1,112 @@
+        """Check that cache is up to date and refresh cache from db.
+        Returns a fresh addresses to decoders mapping.
+        """
+        if should_update_protocol_cache(self.base.database, CacheType.MERKL_REWARD_PROTOCOLS) is True:  # noqa: E501
+            query_merkl_rewards(evm_inquirer=self.evm_inquirer, base_tools=self.base)
+        elif len(self.rewards_protocols) != 0:
+            return None  # we didn't update the globaldb cache, and we have the data already
+
+        with GlobalDBHandler().conn.read_ctx() as cursor:
+            for account in self.base.tracked_accounts.get(self.evm_inquirer.blockchain):
+                for tokens_protocols in globaldb_get_general_cache_values(
+                    cursor=cursor,
+                    key_parts=(
+                        CacheType.MERKL_REWARD_PROTOCOLS,
+                        str(self.evm_inquirer.chain_id.serialize_for_db()),
+                        account,
+                    ),
+                ):
+                    token, protocols = tokens_protocols.split('_')
+                    self.rewards_protocols[account][string_to_evm_address(token)] = protocols.split(',') if protocols else []  # noqa: E501
+
+        return self.addresses_to_decoders()
+
+    def _decode_reward_claim(self, context: DecoderContext) -> DecodingOutput:
+        """Decode a Merkl reward claim event."""
+        if context.tx_log.topics[0] != REWARD_CLAIMED:
+            return DEFAULT_DECODING_OUTPUT
+
+        for event in context.decoded_events:
+            if (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.address == context.tx_log.address and
+                event.asset == (claimed_asset := self.base.get_or_create_evm_asset(
+                    address=(claimed_asset_addr := bytes_to_address(context.tx_log.topics[2])),
+                )) and
+                event.amount == asset_normalized_value(
+                    amount=int.from_bytes(context.tx_log.data[:32]),
+                    asset=claimed_asset,
+                )
+            ):
+                event.event_subtype = HistoryEventSubType.REWARD
+                event.counterparty = CPT_MERKL
+                protocols = self.rewards_protocols[bytes_to_address(context.tx_log.topics[1])].get(claimed_asset_addr, [])  # noqa: E501
+                from_str = f"{', '.join(protocols)} via Merkl" if protocols else 'Merkl'
+                event.notes = f'Claim {event.amount} {claimed_asset.symbol} from {from_str}'
+                break
+        else:
+            log.error(f'Failed to find Merkl reward claim event in {context.transaction}')
+
+        return DEFAULT_DECODING_OUTPUT
+
+    # -- DecoderInterface methods
+
+    def addresses_to_decoders(self) -> dict[ChecksumEvmAddress, tuple[Any, ...]]:
+        return dict.fromkeys([MERKL_DISTRIBUTOR_ADDRESS], (self._decode_reward_claim,))
```

Comment:
if it is just one address you can return the dict difrectly right? You have a list creation, function call and dict creation here